### PR TITLE
Fixes #22534: List Managed Nodes API with any of the fields listed in the "full" never answer

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/22534

Add a methode to be able to filter out entries in a subtree research. 
Then use it in the LDAP "get inventories" to limit the subtrees to the one of the looked for nodes. 
We need a second request to get relevant machine (which use the same id of only requesting machines actually linked to nodes by filtering out the tree)

### CODE REVIEW

File: webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/services/core/FullInventoryRepository.scala [hunk 1]
```diff
@@ -39,6 +39,7 @@ package com.normation.inventory.services.core
 
 import com.normation.errors._
 import com.normation.inventory.domain._
+import zio.Chunk
```
Comment (ncharles) [inline webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/services/core/FullInventoryRepository.scala:42]: this don't seem to be used

File: webapp/sources/ldap-inventory/inventory-repository/src/main/scala/com/normation/inventory/ldap/core/FullInventoryRepositoryImpl.scala [hunk 2]
```diff
@@ -337,6 +338,79 @@ class FullInventoryRepositoryImpl(
...
+        .sorted
+        .map(a => Filter.create(s"entryDN:dnSubtreeMatch:=${a}"))
+      BuildFilter.OR(dnFilters: _*)
+    }
+
+    for {
+      con      <- ldap
+      dit       = inventoryDitService.getDit(inventoryStatus)
+      // Get base tree, we will go into each subtree after
+      nodeTree <- con
```
Comment (ncharles) [inline webapp/sources/ldap-inventory/inventory-repository/src/main/scala/com/normation/inventory/ldap/core/FullInventoryRepositoryImpl.scala:358]: there are several nodes, so naming it nodesTree would help maintenance

**Review by ncharles — CHANGES_REQUESTED** [review]
this is a neat change. Two small cosmetic change, you can self merge after

File: webapp/sources/ldap-inventory/inventory-repository/src/main/scala/com/normation/inventory/ldap/core/FullInventoryRepositoryImpl.scala (hunk 2 above)
Comment (fanf) [reply webapp/sources/ldap-inventory/inventory-repository/src/main/scala/com/normation/inventory/ldap/core/FullInventoryRepositoryImpl.scala:358]: It was to keep the same structure than above method, but the smart move would have been to factor out that actually

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
OK, merging this PR

**Pull request merged**